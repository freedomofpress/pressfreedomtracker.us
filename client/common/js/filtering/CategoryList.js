import React from 'react'

import { GENERAL_ID } from '~/filtering/constants'


function CategoryList({ categories }) {
	return (
		<ul className="filters__summary-list">
			{categories.filter(({ id }) => id !== GENERAL_ID).map(category => (
				<li key={category.id} className="filters__summary-item filters__summary-item--semicolons">
					{category.title}
				</li>
			))}
		</ul>
	)
}


export default CategoryList
