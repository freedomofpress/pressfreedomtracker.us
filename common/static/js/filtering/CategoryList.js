import React from 'react'


function CategoryList({ categories }) {
	return (
		<ul className="filters__summary-list">
			{categories.map(category => (
				<li key={category.id} className="filters__summary-item filters__summary-item--semicolons">
					{category.title}
				</li>
			))}
		</ul>
	)
}


export default CategoryList
