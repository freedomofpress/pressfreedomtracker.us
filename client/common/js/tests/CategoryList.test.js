import React from 'react';
import { shallow, mount } from 'enzyme';
import CategoryList from '~/filtering/CategoryList';
import {
	CATEGORIES_FILTER,
	CATEGORIES_ENABLED,
    FILTER_VALUES,
} from '~/tests/factories/CategoriesFilterFactory';

describe('CategoryList', () => {
    let categories;
    let categoriesEnabled;

    beforeEach(() => {
        categories = CATEGORIES_FILTER;
        categoriesEnabled = CATEGORIES_ENABLED;
    });

    afterEach(() => {
        categories = null;
        categoriesEnabled = null;
    });

    it('should render without error in "debug" mode', () => {
        const component = shallow(
            <CategoryList
                categories={categories.filter(({ id }) => categoriesEnabled[id])}
            />
        );

        expect(component.debug()).toMatchSnapshot();
	});

	it('should render CategoryList properly with data', () => {
		const component = mount(
            <CategoryList
                categories={categories.filter(({ id }) => categoriesEnabled[id])}
            />
        );
        const categoryList = component.find('.filters__summary-item');
		expect(categoryList.text()).toEqual(
			'Leak Case'
		);
    });
});