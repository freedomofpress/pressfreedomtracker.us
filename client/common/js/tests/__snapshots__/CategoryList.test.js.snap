// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList should render without error in "debug" mode 1`] = `
"<ul className=\\"filters__summary-list\\">
  <li className=\\"filters__summary-item filters__summary-item--semicolons\\">
    Leak Case
  </li>
</ul>"
`;
